Documentation/contributing/development/bgp_cplane.rst

```diff
@@ -0,0 +1,99 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+
+.. _bgp_cplane_contributing:
+
+=================
+BGP Control Plane
+=================
+
+This section is specific to BGP Control Plane contributions.
+
+Development Environment
+=======================
+
+BGP Control Plane requires a BGP peer for testing. This section describes a `ContainerLab`_ and `Kind`_-based development environment. The following diagram shows the topology:
+
+.. _ContainerLab: https://containerlab.dev/
+.. _Kind: https://kind.sigs.k8s.io/
+
+.. image:: _static/bgp-cplane-dev-topo.png
+    :align: center
+
+The following describes the role of each node:
+
+* ``router0`` is an `FRRouting (FRR)`_ router. It is pre-configured with minimal peering settings with server0 and server1.
+* ``server0`` and ``server1`` are ``nicolaka/netshoot`` containers that each share a network namespace with their own Kind node.
+* ``server2`` is a non-Cilium ``nicolaka/netshoot`` node useful for testing traffic connectivity from outside of the k8s cluster.
+
+.. _FRRouting (FRR): https://frrouting.org/
+
+Deploy Lab
+----------
+
+.. code-block:: shell-session
+
+   $ make kind-bgp-v4
+
+.. note::
+        Above example setups IPv4 single-stack environment. You can change ``v4`` part to ``v6`` or ``dual`` to setup IPv6 single-stack or dual-stack environment respectively. Same for all below examples.
+
+Install Cilium on the Lab
+-------------------------
+
+Install Cilium on the lab with your favorite way. The following example assumes you are modifying the source and want to build your own image. The minimal mandatory Helm values are provided in ``contrib/containerlab/bgp-cplane-dev-v4/values.yaml``. If needed, you can add Helm values to deploy BGP Control Plane with a different Cilium configuration.
+
+.. code-block:: shell-session
+
+   $ make docker-cilium-image docker-operator-generic-image
+   $ kind --name bgp-cplane-dev-v4 load docker-image quay.io/cilium/cilium:latest quay.io/cilium/operator-generic:latest
+   $ cilium install --chart-directory install/kubernetes/cilium -f contrib/containerlab/bgp-cplane-dev-v4/values.yaml --helm-set image.override="quay.io/cilium/cilium:latest" --helm-set image.pullPolicy=Never --helm-set operator.image.override="quay.io/cilium/operator-generic:latest" --helm-set operator.image.pullPolicy=Never
```

Comment:
This workflow could be simplified by adding the ability to specify the kind cluster name. For example:

```sh
$ git diff Makefile
diff --git a/Makefile b/Makefile
index 3730ee6c01..bd0d692cb0 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,8 @@
 # Copyright Authors of Cilium
 # SPDX-License-Identifier: Apache-2.0

+CLUSTER_NAME ?= kind
+
 ##@ Default
 all: precheck build postcheck ## Default make target that perform precheck -> build -> postcheck
        @echo "Build finished."
@@ -576,7 +578,7 @@ kind-build-image-agent: ## Build cilium-dev docker image

 $(eval $(call KIND_ENV,kind-image-agent))
 kind-image-agent: kind-ready kind-build-image-agent ## Build cilium-dev docker image and import it into kind.
-       $(QUIET)kind load docker-image $(LOCAL_AGENT_IMAGE)
+       $(QUIET)kind load docker-image $(LOCAL_AGENT_IMAGE) -n $(CLUSTER_NAME)

 $(eval $(call KIND_ENV,kind-build-image-operator))
 kind-build-image-operator: ## Build cilium-operator-dev docker image
@@ -584,7 +586,7 @@ kind-build-image-operator: ## Build cilium-operator-dev docker image

 $(eval $(call KIND_ENV,kind-image-operator))
 kind-image-operator: kind-ready kind-build-image-operator ## Build cilium-operator-dev docker image and import it into kind.
-       $(QUIET)kind load docker-image $(LOCAL_OPERATOR_IMAGE)
+       $(QUIET)kind load docker-image $(LOCAL_OPERATOR_IMAGE) -n $(CLUSTER_NAME)

 $(eval $(call KIND_ENV,kind-build-clustermesh-apiserver))
 kind-build-clustermesh-apiserver: ## Build cilium-clustermesh-apiserver docker image
```

A user would then:

```sh
$ CLUSTER_NAME=bgp-cplane-dev-v4 make kind-image
$ cilium install --chart-directory install/kubernetes/cilium -f contrib/containerlab/bgp-cplane-dev-v4/values.yaml
```

Reply:
Following the workflow of the `make kind-xxx` target sounds good to me. I put your suggestion into a separate commit and updated the document. Please take a look.